go.mod

```diff
@@ -3,8 +3,8 @@ module github.com/aws-controllers-k8s/applicationautoscaling-controller
 go 1.14
 
 require (
-	github.com/aws-controllers-k8s/runtime v0.2.1
-	github.com/aws/aws-sdk-go v1.37.4
+	github.com/aws-controllers-k8s/runtime v0.12.0
```

Comment:
@mbaijal I would recommend to do the runtime update in a separate PR, since will require regenerating all the apis and tuning/re-running all the unit tests

Reply:
Agree. It's quite difficult to see in this review what is simply updating to 0.12.0 and what is custom code. Especially considering we are coming from 0.2.1 - there are a lot of rebasing changes.

Reply:
I had to update the runtime here since I was using the latest code generator changes for this PR which includes late initializers and without this version of runtime things don't work. 

I could however have a separate PR for runtime update which gets merged before this one.

Reply:
Yes please. A separate PR for runtime from v0.2.1 to v0.12.0 will greatly reduce subsequent PRs (like this one)